api/src/infrastructure/postgres-survey-data.repository.ts

```diff
@@ -46,16 +46,28 @@ export class PostgresSurveyDataRepository implements ISurveyDataRepository {
     return sections;
   }
 
+  public async addSurveyDataToBaseWidget(
+    widget: BaseWidgetWithData,
+    filters?: WidgetDataFilters,
+  ): Promise<BaseWidgetWithData> {
+    let filterClause: string;
+    if (filters !== undefined) {
+      filterClause = this.sqlAdapter.generateSqlFromWidgetDataFilters(filters);
+    }
+
+    await this.appendBaseWidgetData(widget, filterClause);
+    return widget;
+  }
+
   private async appendBaseWidgetData(
-    answersTable: string,
-    filterClause: string,
     widget: BaseWidgetWithData,
+    filterClause: string,
   ): Promise<void> {
     const { indicator } = widget;
     // Edge cases here. Total surveys and total countries.
     if (indicator === 'total-surveys') {
-      const filteredCount = `SELECT COUNT(count)::integer as count FROM (SELECT COUNT(DISTINCT survey_id) FROM ${answersTable} ${filterClause} GROUP BY survey_id) AS survey_count`;
-      const totalCount = `SELECT COUNT(count)::integer as count FROM (SELECT COUNT(DISTINCT survey_id) FROM ${answersTable} GROUP BY survey_id) AS survey_count`;
+      const filteredCount = `SELECT COUNT(count)::integer as count FROM (SELECT COUNT(DISTINCT survey_id) FROM ${this.answersTable} ${filterClause} GROUP BY survey_id) AS survey_count`;
```

Comment:
Again, we agreed on deprecating this approach of using raw strings to perform queries, which adds maintainability overhead, and lacks typings